src/main/node/config-manager.test.ts

```diff
@@ -1,21 +1,266 @@
+    expect([
+      oldConfig.publicationName,
+      oldConfig.description,
+      oldConfig.collaborators,
+    ]).toEqual([
+      updatedConfig.publicationName,
+      updatedConfig.description,
+      updatedConfig.collaborators,
+    ]);
+  });
+});
 
+describe('updateCollaborators', () => {
+  it('collaborators property in the config is correctly updated, while other fields remain unchanged', () => {
+    const updatedCollaborators = [
+      { id: 'id', role: 'role', githubUsername: 'githubUsername' },
+    ];
+    const oldConfig = prepareNewConfigFile();
+    const updatedConfig = updateConfigField(
+      nodeDirectoryPath,
+      updatedCollaborators,
+      updateCollaborators
+    );
+    expect(updatedConfig.collaborators).toEqual(updatedCollaborators);
+    expect([
+      oldConfig.publicationName,
+      oldConfig.description,
+      oldConfig.packageManager,
+    ]).toEqual([
+      updatedConfig.publicationName,
+      updatedConfig.description,
+      updatedConfig.packageManager,
+    ]);
+  });
+});
+
+describe('updateTag', () => {
+  it('tag property in the config is correctly updated, while other fields remain unchanged', () => {
+    const updatedTag = '08/11/2021 7.37PM';
+    const oldConfig = prepareNewConfigFile();
+    const updatedConfig = updateConfigField(
+      nodeDirectoryPath,
+      updatedTag,
+      updateTag
+    );
+    expect(updatedConfig.tag).toEqual(updatedTag);
+    expect([
+      oldConfig.publicationName,
+      oldConfig.description,
+      oldConfig.packageManager,
+    ]).toEqual([
+      updatedConfig.publicationName,
+      updatedConfig.description,
+      updatedConfig.packageManager,
+    ]);
+  });
+});
+
+describe('updateName', () => {
+  it('name property cannot be empty or null', () => {
     expect(() => {
-      fs.readFileSync('./src/main/node/vn-pub.conf');
-    }).not.toThrow();
+      tryToUpdateWithInvalidParam(nodeDirectoryPath, '', updateName);
+    }).toThrowError();
+    expect(() => {
+      tryToUpdateWithInvalidParam(nodeDirectoryPath, null, updateName);
+    }).toThrowError();
+  });
+});
+
+describe('updatePackageManager', () => {
+  it('packageManager property cannot be empty or null', () => {
+    expect(() => {
+      tryToUpdateWithInvalidParam(nodeDirectoryPath, '', updatePackageManager);
+    }).toThrowError();
+    expect(() => {
+      tryToUpdateWithInvalidParam(
+        nodeDirectoryPath,
+        null,
+        updatePackageManager
+      );
+    }).toThrowError();
   });
 });
+
+describe('updateCollaborators', () => {
+  it('collaborators property cannot be empty or null', () => {
+    createConfigFile(nodeDirectoryPath, config);
+    expect(() => {
+      tryToUpdateWithInvalidParam(nodeDirectoryPath, null, updateCollaborators);
+    }).toThrowError();
+    expect(() => {
+      tryToUpdateWithInvalidParam(nodeDirectoryPath, [], updateCollaborators);
+    }).toThrowError();
+  });
+});
```

Comment:
With the simplification of update functions to just one `updateconfigFunction` the tests also need refactoring. They can become more concise and readable with less code obstructing the meaning of the test. To start it off this would be the three main overloads tested:

```ts
describe('updateConfigField', () => {
  beforeEach(() => createConfigFile(nodeDirectoryPath, config));

  it('correctly updates the publicationName', () => {
    updateConfigField(nodeDirectoryPath, 'publicationName', 'newName');
    expect(getConfigFile(nodeDirectoryPath).publicationName).toBe('newName');
  });

  it('correctly updates useTypescript', () => {
    updateConfigField(nodeDirectoryPath, 'useTypescript', false);
    expect(getConfigFile(nodeDirectoryPath).useTypescript).toBe(false);
  });

  it('correctly updates collaborators', () => {
    updateConfigField(nodeDirectoryPath, 'collaborators', [
      { githubUsername: 'test', id: 'test', role: 'test' },
    ]);
    expect(getConfigFile(nodeDirectoryPath).collaborators).toEqual([
      { githubUsername: 'test', id: 'test', role: 'test' },
    ]);
  });
});
```